<a href="https://colab.research.google.com/github/Swasthi28/industryconnect/blob/main/heart_disease.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
df = pd.read_csv("https://talentcocomedia.s3.amazonaws.com/ml-assets/heart-disease.csv")
df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [3]:
df.tail()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0

# Research

The dataset available at "https://talentcocomedia.s3.amazonaws.com/ml-assets/heart-disease.csv" contains various features associated with heart disease, including:


1. Age: The age of the patient in years (numeric)
2. Sex: The sex of the patient (1 = male; 0 = female)
3. Chest pain type (cp): The type of chest pain experienced by the patient (1 = typical angina, 2 = atypical angina, 3 = non-anginal pain, 4 = asymptomatic)
4. Resting blood pressure (trestbps): The resting blood pressure of the patient (in mm Hg on admission to the hospital) (numeric)
5. Serum cholesterol (chol): The serum cholesterol level of the patient (in mg/dl) (numeric)
6. Fasting blood sugar (fbs): The fasting blood sugar level of the patient (> 120 mg/dl or not) (1 = true; 0 = false)
7. Resting electrocardiographic results (restecg): Results of the resting electrocardiogram (0 = normal, 1 = having ST-T wave abnormality, 2 = showing probable or definite left ventricular hypertrophy)
8. Maximum heart rate achieved (thalach): The maximum heart rate achieved during the exercise test (numeric)
9. Exercise-induced angina (exang): Whether or not the patient had exercise-induced angina (1 = yes; 0 = no)
10. ST depression induced by exercise relative to rest (oldpeak): ST depression induced by exercise relative to rest (numeric)
11. Slope of the peak exercise ST segment (slope): The slope of the peak exercise ST segment (1 = upsloping, 2 = flat, 3 = downsloping)
12. Number of major vessels (0-3) colored by fluoroscopy (ca): The number of major vessels (0-3) colored by fluoroscopy (numeric)
Thal: A blood disorder called thalassemia (3 = normal, 6 = fixed defect, 7 = reversable defect)
13. Target variable (target): Whether or not the patient has heart disease (1 = yes; 0 = no)

Each row in the dataset represents a patient, and each column represents a feature associated with the patient's heart disease status.

In [4]:
df.shape

(303, 14)